### PR TITLE
DynamoDB Indexing and Data Types

### PR DESCRIPTION
**Pull request by kimschmidtsbrain**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1020,14 +1020,15 @@ DynamoDB
 -	There is a [local version](https://aws.amazon.com/blogs/aws/dynamodb-local-for-desktop-development/) of DynamoDB provided for developer use.
 -	[DynamoDB Streams](http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Streams.html) provides an ordered stream of changes to a table. Use it to replicate, back up, or drive events off of data
 -	DynamoDB can be used [as a simple locking service](https://gist.github.com/ryandotsmith/c95fd21fab91b0823328).
+-       DynamoDB indexing can include primary keys, which can either be a single-attrubute hash key or a composite hash-key range. You can also query non-primary key attributes using [Secondary Indexes](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/SecondaryIndexes.html).
```
Comment (jlevy) [inline README.md]: Nit: Boldface **primary keys**
Nit: capitalization and boldface on secondary indexes
> ... using [**secondary indexes**](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/SecondaryIndexes.html).

File: README.md [hunk 2]
```diff
@@ -1020,14 +1020,15 @@ DynamoDB
...
 -	🔸 Dealing with **time series data** in DynamoDB can be challenging. A global secondary index together with down sampling timestamps can be a possible solution as explained [here](https://blogs.aws.amazon.com/bigdata/post/Tx3KPZDXIBJEQ4B/Scaling-Writes-on-Amazon-DynamoDB-Tables-with-Global-Secondary-Indexes).
 -	DynamoDB does [not allow](https://forums.aws.amazon.com/thread.jspa?threadID=90137) an empty string as a valid attribute value. The most common work-around is to use a substitute value instead of leaving the field empty.
-
+-       DynamoDB supports three [data types](http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBMapper.DataTypes.html) - number, string, and binary - in both scalar and multi-valued sets. DynamoDB can also support [JSON](https://aws.amazon.com/blogs/aws/dynamodb-update-json-and-more/).
```
Comment (jlevy) [inline README.md]: Nits: Add a heading here. It's a sad battle to fight, but we also try to use em dash:

> **Data types:** DynamoDB supports three [data types](http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBMapper.DataTypes.html) — **number**, **string,** and **binary** — in both scalar and multi-valued sets. DynamoDB can also support [JSON](https://aws.amazon.com/blogs/aws/dynamodb-update-json-and-more/).

**Comment by kimschmidtsbrain** [comment]
Ok, are you asking for another DynamoDB topic? Lynn & I started beers at noon...but I'm home now so can be done

File: README.md (hunk 1 above)
Comment (ThanosBaskous) [inline README.md]: Typo: "attrubute" - "attribute"

File: README.md [hunk 3]
```diff
@@ -1020,14 +1020,15 @@ DynamoDB
...
 -	🔸 Dealing with **time series data** in DynamoDB can be challenging. A global secondary index together with down sampling timestamps can be a possible solution as explained [here](https://blogs.aws.amazon.com/bigdata/post/Tx3KPZDXIBJEQ4B/Scaling-Writes-on-Amazon-DynamoDB-Tables-with-Global-Secondary-Indexes).
 -	DynamoDB does [not allow](https://forums.aws.amazon.com/thread.jspa?threadID=90137) an empty string as a valid attribute value. The most common work-around is to use a substitute value instead of leaving the field empty.
-
+-       Data Types: DynamoDB supports three [data types](http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBMapper.DataTypes.html) -- **number**, **string**, and **binary** -- in both scalar and multi-valued sets. DynamoDB can also support [JSON](https://aws.amazon.com/blogs/aws/dynamodb-update-json-and-more/).
```
Comment (ThanosBaskous) [inline README.md]: Non-blocking nit: em-dash preferred over double hyphen.

**Comment by ThanosBaskous** [comment]
I'm going to merge and fix the non-blocking nits in a separate PR. Since the changes appear to be the same as the ones in #250, I'll close that one.

Thanks @kimschmidtsbrain!

**Comment by jlevy** [comment]
👍 Thanks so much!

**Comment by kimschmidtsbrain** [comment]
Well, first day & I learned a whole lot of what NOT to do on GitHub. I hope
everything is cool now with my submissions.

Sincerely,

Kim Schmidt
949-735-8505
@DataLeader, @WearableQueen
https://www.linkedin.com/in/kimschmidt1

**Pull request merged**